src/map.cpp

```diff
@@ -1180,6 +1180,14 @@ bool Map::isBlockOccluded(MapBlock *block, v3s16 cam_pos_nodes)
 	// this is a HACK, we should think of a more precise algorithm
 	u32 needed_count = 2;
 
+	v3s16 random_point(myrand_range(-bs2, bs2), myrand_range(-bs2, bs2), myrand_range(-bs2, bs2));
```

Comment:
a more predictable approach would be to pick one of the 9 or 10 existing points and rotate around them.
random is probably fine in practice but with round-robin you have a worst-case time until all relevant corners have definitely been checked

perhaps `gametime % 9` would even work for that (not sure how often this runs)

Reply:
In an earlier version I had used dir9[myrand()%9].
I like this better because it will eventually try more points, specifically points inside the block. The longer it runs the more accurate it gets.

I even tried setting the cull_optimize distance to 2, just to see what happens. It pretty quickly arrives at scene with no missing blocks.